core/src/main/java/io/kestra/core/tasks/flows/Flow.java

```diff
@@ -149,7 +148,7 @@ public Optional<WorkerTaskResult> createWorkerTaskResult(
         io.kestra.core.models.flows.Flow flow,
         Execution execution
     ) {
-        // we only create a worker task result when the taskrun is terminated
+        // we only create a worker task result when the execution is terminated
```

Comment:
I think taskrun is more appropriated as we literally  check for the taskrun state ? 
Do I miss something ?

Reply:
Well, this is a bit convoluted but it's the subflow execution termination that trigger the creation of a worker task result to terminate the parent taskrun.
But there is a trick with the parent taskrun to convey some state ..
Anyway, there is a refactoring ongoing that will clarify this.